diff --git a/acpi_call.c b/acpi_call.c
index 3025d97..fd809b9 100644
--- a/acpi_call.c
+++ b/acpi_call.c
@@ -6,7 +6,7 @@
 #include <linux/proc_fs.h>
 #include <linux/slab.h>
 #include <asm/uaccess.h>
-#include <acpi/acpi.h>
+#include <linux/acpi.h>

 MODULE_LICENSE("GPL");

@@ -317,10 +317,9 @@ static ssize_t acpi_proc_read( struct file *filp, char __user *buff,
     return ret;
 }

-static struct file_operations proc_acpi_operations = {
-        .owner    = THIS_MODULE,
-        .read     = acpi_proc_read,
-        .write    = acpi_proc_write,
+static const struct proc_ops proc_acpi_operations = {
+        .proc_read  = acpi_proc_read,
+        .proc_write = acpi_proc_write,
 };

 #else
